src/components/tire/AddTireCheckPage.js

```diff
@@ -0,0 +1,257 @@
+        bottomRight: globalTirePressure,
+      };
+      setCurrentTirePressure(newTirePressure);
+      setInitialTirePressure(newTirePressure);
+      setInitialGlobalTirePressure(globalTirePressure);
+    });
+  };
+
+  const resetTirePressure = () => {
+    setCurrentTirePressure(initialTirePressure);
+    setGlobalTirePressure(initialGlobalTirePressure);
+  };
+
+  const hasChanged = () => {
+    const hasChanged =
+      currentTirePressure !== initialTirePressure ||
+      globalTirePressure !== initialGlobalTirePressure;
+    return hasChanged;
+  };
+
+  return (
+    <>
+      <Breadcrumb
+        uriSegments={[
+          BreadcrumbIndicies.home,
+          BreadcrumbIndicies.tires,
+          BreadcrumbIndicies.add,
+        ]}
+      />
+
+      <h2>New tire pressure</h2>
+      <div className={styles.newTirePressure}>
+        <TirePressure
+          customClass={styles.globalTireIncrementer}
+          numericValue={globalTirePressure}
+          callback={setGlobalTirePressure}
+        />
+      </div>
+
+      <h2 className={styles.previousTireHeading}>Previous tire pressure</h2>
+      <div className={styles.tireGrid}>
+        <TirePressure
+          label="Top Left Tire"
+          customClass={styles.topLeftTireIncrementer}
+          numericValue={currentTirePressure[TirePositions.topLeft]}
+          callback={(value) => setCurrentTire(value, TirePositions.topLeft)}
+        />
+        <TirePressure
+          label="Top Right Tire"
+          customClass={styles.topRightTireIncrementer}
+          numericValue={currentTirePressure[TirePositions.topRight]}
+          callback={(value) => setCurrentTire(value, TirePositions.topRight)}
+        />
+        <TirePressure
+          label="Bottom Left Tire"
+          customClass={styles.bottomLeftTireIncrementer}
+          numericValue={currentTirePressure[TirePositions.bottomLeft]}
+          callback={(value) => setCurrentTire(value, TirePositions.bottomLeft)}
+        />
+        <TirePressure
+          label="Bottom Right Tire"
+          customClass={styles.bottomRightTireIncrementer}
+          numericValue={currentTirePressure[TirePositions.bottomRight]}
+          callback={(value) => setCurrentTire(value, TirePositions.bottomRight)}
+        />
+      </div>
+
+      {hasChanged() ? (
+        <>
+          <button
+            className={[
+              sharedButtons.button,
+              styles.saveTirePressureButton,
+            ].join(" ")}
+            onClick={handleSave}
+          >
+            Add tire check
+          </button>
+          <button
+            className={[
+              sharedButtons.button,
+              styles.resetTirePressureButton,
+            ].join(" ")}
+            onClick={resetTirePressure}
+          >
+            Reset tire check
+          </button>
+        </>
+      ) : (
+        ""
+      )}
+    </>
+  );
+}
+
+function TirePressure({ numericValue, callback, customClass, label }) {
```

Comment:
This is the third type of iteration for this type of component, where this time you can actually push the callbacks down deeper into the component.

I am thinking that this might be a better way to handle these types of things.